src/components/drBidPage/submit/addBid.tsx

```diff
@@ -22,48 +29,101 @@ const AddBid: React.FC = () => {
   // price
   const [price, setPrice] = useState<number | undefined>(undefined);
 
-  // price input disable
-  const [priceDisable, setPriceDisable] = useState<boolean>(false);
-
   // set submit button disabled
   const [submitDisabled, setSubmitDisabled] = useState<boolean>(true);
 
+  // determine interval array by selected month
+  useEffect(() => {
+    if (date) {
+      const mth = dayjs(date).get('month');
+      if (mth >= 5 && mth <= 8) {
+        setIntervalArr(['23:00 - 8:00', '8:00 - 18:00', '18:00 - 23:00']);
+      } else {
+        setIntervalArr(['22:00 - 17:00', '17:00 - 22:00']);
+      }
+    }
+  }, [date]);
+
   // determine data validity
   useEffect(() => {
-    if (date && mode !== 0 && volume && volume !== 0 && price && price !== 0) {
+    if (
+      date &&
+      interval !== '' &&
+      mode !== 0 &&
+      volume &&
+      volume !== 0 &&
+      price &&
+      price !== 0
+    ) {
       setSubmitDisabled(false);
     } else {
       setSubmitDisabled(true);
     }
-  }, [date, mode, volume, price]);
+  }, [date, interval, mode, volume, price]);
 
   // determine price by mode
   useEffect(() => {
     switch (mode) {
-      default:
-        break;
-      case 1: {
-        setPriceDisable(true);
+      default: {
         const hr = dayjs().get('hour');
```

Comment:
You should use the selected `interval` but not the current hour.
It may be something like this:
```
if (interval === '23:00 - 8:00') {
  setPrice(...);
} else if (interval === '8:00 - 18:00') {
  setPrive(...);
} else {
  setPrice(...);
}
```

Reply:
done thx